test/escort.test.js

```diff
@@ -15,1586 +15,1702 @@ var makeBadString = (function (Ctor) {
     };
 }(String));
 
-module.exports = {
-    "methods static": function () {
-        var url;
-        var app = connect(
-            escort(function (routes) {
-                url = routes.url;
-                
-                methods.forEach(function (method) {
-                    routes[method]("home_" + method, "/" + method, function (req, res) {
-                        res.end(method.toUpperCase() + " /" + method);
-                    });
-                });
-            })
-        );
-        
+function makeConnect(...apps) {
+  const c = connect();
+
+  for (let app of apps) {
+    c.use(app);
+  }
+
+  return c;
+}
+
+describe("escort", function() {
+  after(function() {
+    // apps leave references that don't get cleaned up
+    process._getActiveHandles()
+      .filter(h => h instanceof http.Server)
+      .forEach(h => h.unref());
+  });
+
+  it("methods static", async function() {
+    var url;
+    var app = makeConnect(
+      escort(function (routes) {
+        url = routes.url;
+
         methods.forEach(function (method) {
-            assert.response(app,
-                { url: "/" + method, method: method.toUpperCase() },
-                { body: method.toUpperCase() + " /" + method });
-            
-            assert.strictEqual("/" + method, url["home_" + method]());
-
-            methods.forEach(function (otherMethod) {
-                if (method !== otherMethod) {
-                    assert.response(app,
-                        { url: "/" + method, method: otherMethod.toUpperCase() },
-                        { statusCode: 405 });
-                }
-            });
+          routes[method]("home_" + method, "/" + method, function (req, res) {
+            res.end(method.toUpperCase() + " /" + method);
+          });
         });
-    },
-    "bind static": function () {
-        var url;
-        var app = connect(
-            escort(function (routes) {
-                url = routes.url;
-                
-                var descriptor = {};
-                methods.forEach(function (method) {
-                    descriptor[method] = function (req, res) {
-                        res.end(method.toUpperCase() + " /");
-                    };
-                });
-                routes.bind("home", "/", descriptor);
-            })
-        );
-        
-        assert.strictEqual("/", url.home());
-        
+      })
+    );
+
+    for (let method of methods) {
+      await assert.response(app,
+                            { url: "/" + method, method: method.toUpperCase() },
+                            { body: method.toUpperCase() + " /" + method });
+
+      assert.strictEqual("/" + method, url["home_" + method]());
+
+      for(let otherMethod of methods) {
+        if (method !== otherMethod) {
+          await assert.response(app,
```

Comment:
While nothing about the escort router itself is particularly async, _supertest is_, so we need to wait for it to finish. This involved converting a bunch of `forEach` statements to `for .. of` loops.